Assets/IntegrationTests/StoppableWaitForTime.cs

```diff
@@ -0,0 +1,41 @@
+﻿using UnityEngine;
+using System.Collections;
+
+namespace Shopify.Unity.Tests {
+    public class StoppableWaitForTime : CustomYieldInstruction {
+        public override bool keepWaiting {
+            get {
+                return !_IsStopped && Time.time - StartTime < _MaxTime;
+            }
+        }
+
+        public float DurationToStop {
+            get {
+                return StopTime - StartTime;
+            }
+        }
+
+        public bool IsStopped {
+            get {
+                return _IsStopped;
+            }
+        }
+
+        float _MaxTime;
+        bool _IsStopped;
+
+        float StopTime;
+        float StartTime;
```

Comment:
Nit: Should these be private?

Reply:
Not sure you should be able to modify them because its just doing bad things if you do.

Reply:
Gah sorry... Just realize I read your comment wrong. They are private by default.

Reply:
they should follow the naming convention for private member variables though (underscore prefix)